ui/pages/permissions-connect/permissions-connect.component.js

```diff
@@ -175,8 +177,16 @@ export default class PermissionConnect extends Component {
   }
 
   componentDidUpdate(prevProps) {
-    const { permissionsRequest, lastConnectedInfo, origin } = this.props;
-    const { redirecting } = this.state;
+    const { permissionsRequest, lastConnectedInfo, targetSubjectMetadata } =
+      this.props;
+    const { redirecting, origin } = this.state;
+
+    if (
```

Comment:
```suggestion
    // We cache the last known good targetSubjectMetadata since it may be null when the approval is cleared
    if (
```
IMO this deserves a comment

Reply:
Updated here https://github.com/MetaMask/metamask-extension/pull/29880/commits/4f47a967f20c391723947160800bb612a1b9505b